node-api/src/main/kotlin/net/corda/nodeapi/internal/persistence/CordaPersistence.kt

```
@@ -266,4 +268,17 @@ private fun Throwable.hasSQLExceptionCause(): Boolean =
             else -> cause?.hasSQLExceptionCause() ?: false
         }
 
-class CouldNotCreateDataSourceException(override val message: String?, override val cause: Throwable? = null) : Exception()
\ No newline at end of file
+class CouldNotCreateDataSourceException(override val message: String?, override val cause: Throwable? = null) : Exception()
+
+private fun requireCorrectAttachmentsContractsTableName(connection: Connection) {
+    val correctName = "NODE_ATTACHMENTS_CONTRACTS"
+    val incorrectV30Name = "NODE_ATTACHMENTS_CONTRACT_CLASS_NAME"
+    val incorrectV31Name = "NODE_ATTCHMENTS_CONTRACTS"
+
+    fun warning(incorrectName: String) = "The database contains the older table name $incorrectName instead of $correctName,  migrate your database as per Upgrade Notes."
+
+    if (!connection.metaData.getTables(null, null, correctName, null).next()) {
+        require(!connection.metaData.getTables(null, null, incorrectV30Name, null).next()) { warning(incorrectV30Name) }
```

You can throw a custom exception here and catch it in NodeStartup to print a nice message 
Also I would add the link to the actual upgrade notes